performant-software/Annotation-Studio
# Made changes to several UI elements

### What does this PR do?

- Makes group nomenclature configurable
- Adds snapshot preview
- Adjusts document phases
- Makes typekit configurable
- Makes navbar brand configurable

Closes #28
Closes #39 
Closes #37

### How to test

This PR is staged for review at: https://cove-studio-pr-40.herokuapp.com/ (note that the "View Deployment" button in the conversation thread below points there too).

You will need an account, with the role of teacher or admin, to see all features. Shoot me an email when you've created an account, and I will add those roles.

- Document phases: 
  1. Visit: https://cove-studio-pr-40.herokuapp.com/dashboard
  1. Click create new document (https://cove-studio-pr-40.herokuapp.com/documents/new)
  1. Note that available document phases are now: draft/annotatable/published/archived. *If you don't see those phases, it's likely due to your not having teacher or admin role. Let me know and I will fix that*.

- Group nomenclature:
  1. Visit: https://cove-studio-pr-40.herokuapp.com/dashboard and https://cove-studio-pr-40.herokuapp.com/users/edit
  1. Note that in both places, instead of being called "Classes" and "Groups", those entities are called "Covens" and "Cabals", just to demonstrate configurability.

- Snapshot preview: 
  1. Visit https://cove-studio-pr-40.herokuapp.com/documents/example
  1. Annotate. 
  1. Open the annotation filter sidebar, and click snapshot at the bottom.
  1. Visit https://cove-studio-pr-40.herokuapp.com/documents/example/preview
  1. Note that your annotations are visible.

- Typekit configurability
  1. In one browser window, open https://cove-studio-pr-40.herokuapp.com/documents/example/ 
  1. In another, open: https://cove-studio.herokuapp.com/documents/example/
  1. Note that the typeface used for the main document text on the first is different from that used on the second.
  1. Hit Option-Command-J on the keyboard to open the javascript console. Note that there is not an error regarding typekit not being enabled for this domain.

- Brand configurability
  1. In one browser window, visit: http://mit.annotationstudio.org; take note of the branding of the application at the top of all pages: Annotation Studio MIT (in MIT red color).
  1. In another browser window, visit: https://cove-studio-pr-40.herokuapp.com/; take note of the branding of the application at the top of all pages: NAVSA ... COVE Editions ... Cove Studio.
